src/Security/TokenAuthenticator.php

```diff
@@ -0,0 +1,165 @@
+<?php
+/**
+ * @file
+ * Token authentication using adgangsplatform introspection end-point.
+ */
+
+namespace App\Security;
+
+use Symfony\Component\DependencyInjection\ParameterBag\ParameterBagInterface;
+use Symfony\Component\HttpFoundation\JsonResponse;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\Security\Core\Authentication\Token\TokenInterface;
+use Symfony\Component\Security\Core\Exception\AuthenticationException;
+use Symfony\Component\Security\Core\User\UserInterface;
+use Symfony\Component\Security\Core\User\UserProviderInterface;
+use Symfony\Component\Security\Guard\AbstractGuardAuthenticator;
+use Symfony\Contracts\HttpClient\Exception\ExceptionInterface;
+use Symfony\Contracts\HttpClient\Exception\HttpExceptionInterface;
+use Symfony\Contracts\HttpClient\HttpClientInterface;
+
+/**
+ * Class TokenAuthenticator.
+ */
+class TokenAuthenticator extends AbstractGuardAuthenticator
+{
+    private $client;
+
+    private $clientId;
+    private $clientSecret;
+    private $endPoint;
+
+    /**
+     * TokenAuthenticator constructor.
+     *
+     * @param ParameterBagInterface $params
+     * @param HttpClientInterface $httpClient
+     */
+    public function __construct(ParameterBagInterface $params, HttpClientInterface $httpClient)
+    {
+        $this->client = $httpClient;
+
+        $this->clientId = $params->get('openPlatform.id');
+        $this->clientSecret = $params->get('openPlatform.secret');
+        $this->endPoint = $params->get('openPlatform.introspection.url');
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function supports(Request $request)
+    {
+        return $request->headers->has('authorization');
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getCredentials(Request $request)
+    {
+        return $request->headers->get('authorization');
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getUser($credentials, UserProviderInterface $userProvider)
+    {
+        if (null === $credentials) {
+            // The token header was empty, authentication fails with HTTP Status
+            // Code 401 "Unauthorized"
+            return null;
+        }
+
+        // Parse token information from the bearer authorization header.
+        preg_match('/Bearer\s(\w+)/', $credentials, $matches);
+        if (2 !== count($matches)) {
+            return null;
+        }
+
+        $token = $matches[1];
+
+        try {
+            $response = $this->client->request('POST', $this->endPoint.'?access_token='.$token, [
```

Comment:
Token has `"expires": "3600"` according to https://github.com/DBCDK/hejmdal/blob/master/docs/oauth2.md#5-introspection. Consider caching result for that duration?

Reply:
The actual token from the library has 1 month.... according to the information retrieved from the end-point.

Reply:
Use the value from the token/endpoint. If we use a [cache pool](https://symfony.com/doc/4.4/reference/configuration/framework.html#pools) e.g. `cache.auth` specifically for auth we can always do `cache:pool:clear` to invalidate cached tokens

Reply:
Fixed